bf.c

```diff
@@ -0,0 +1,107 @@
+#include <err.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define CELLS_LEN 30000
+
+void interp(char *prog, char *prog_end, char *cells, char *cells_end) {
+    char *instr = prog;
+    char *cell = cells;
+    while (instr < prog_end) {
+        switch (*instr) {
+            case '>': {
+                if (cell++ == cells_end)
+                    errx(1, "out of memory");
+                instr++;
```

Comment:
every bytecode ends in `instr++` so put it outside?

Reply:
The plan is that a later version will use computed gotos, at which point it can't be moved outside. But, perhaps, that's not very relevant here and now. Thoughts?

Reply:
I don't feel strongly. It's clear either way.

Reply:
On reflection, I think minimising things is a good idea. Fixed in 15d6baf.